Angular/Angular guidelines and best practices/Assets & caching.md

```diff
@@ -0,0 +1,172 @@
+
+This caching mechanism could also be applied to JS and CSS chunks in a way that would not require fingerprinting those files, but since Angular already fingerprints those files, we will leave it as-is.
+
+_One small note_: if you use relative paths in SCSS, for example for a background image, then that image will get fingerprinted. However, if you have an image element in your template, like `<img src="/assets/logo.png">`, the `logo.png` file will not be fingerprinted and you are at risk of the client using a stale version of the file! Please check out [Loading assets via SCSS](#loading-assets-via-scss) sub-chapters that covers this topic.
+
+**Rule #3**: Implement cache re-validation using ETags for all static assets!
+
+## Cache re-validation implementation
+
+Depending on how the application files are served, implementation details for setting up HTTP caching will be a bit different.
+
+We will cover some of the common HTTP serving technologies, including static file serving and Server-side rendering.
+
+### Static file serving
+
+If the application files are served statically, the static file server must be configured to serve and cache assets files correctly.
+
+#### Nginx example
+
+The Nginx configuration file for the site should be updated to enable cache re-validation for the `assets/` directory:
+
+```bash
+    server {
+        listen 80 default_server;
+        root /var/www;
+        location / {
+            add_header Surrogate-Control "public, max-age=3600";
+            add_header Cache-Control "public, max-age=3600";
+            try_files $uri $uri/ /index.html;
+        }
+        location /assets {
+            add_header Surrogate-Control "public, max-age=0";
+            add_header Cache-Control "public, max-age=0";
+            add_header Cache-Control must-revalidate;
+        }
+    }
+```
+
+### Server-side rendering
+
+If the application uses Server-side rendering, then the caching should be handled by the Node.js server.
+
+When using Angular Universal, `server.ts` includes a piece of code in charge of serving the application client build files (this includes `index.html`, JS and CSS chunks, and files from `assets/`). This piece of code should be updated to force cache re-validation:
+
+```typescript
+server.get('*.*', express.static(distFolder, {
+  maxAge: '0',
+  etag: true,
+}));
+```
+
+This change will actually force re-validation for `index.html`, JS and CSS chunks as well, not only the `assets/` directory. This is ok, you could even disable output hashing of chunk files.
```

Comment:
>you could even disable output hashing of chunk files

Could or should?

Reply:
If all chunks are re-validated, you could theoretically turn off output hashing, but there is no harm in leaving it on.

Reply:
Yes, but in the context of the handbook, we should be explicit about what to do.

Reply:
Ok, I will remove that sentence